analyzer/queries/queries.go

```diff
@@ -378,15 +378,27 @@ var (
 
 	RuntimeEVMContractInsert = `
     INSERT INTO chain.evm_contracts
-      (runtime, contract_address, creation_tx, creation_bytecode)
-    VALUES ($1, $2, $3, $4)`
+      (runtime, contract_address, creation_tx, creation_bytecode, gas_used)
+    VALUES ($1, $2, $3, $4, $5)`
 
 	RuntimeEVMContractRuntimeBytecodeUpsert = `
-    INSERT INTO chain.evm_contracts(runtime, contract_address, runtime_bytecode)
-    VALUES ($1, $2, $3)
+    WITH
+      contract_gas_used AS (
+        SELECT SUM(gas_used) AS total_gas
+        FROM chain.runtime_transactions
+        WHERE runtime = $1 AND "to" = $2::text
+      )
```

Comment:
Yeah I would've preferred to avoid this since I'm worried about the performance - I'll do some experiments with the added index before merging. 

> this upsert will always be an update and never an insert

IIRC there are actually a good number of contracts where this is an insert on maininet. Not sure if that's because we started indexing late and missed the `evm.Create` tx, or something else... but in any case it looks like we'll need to support it.

```
oasisindexer=> select count(*) from chain.evm_contracts where creation_tx is null and runtime_bytecode is not null;
 count
-------
   714
(1 row)
```

Reply:
Updated the PR description with timings; for the few contracts that are _very_ popular the upsert (insert) can take a while, but I think it's fine since 1) these updates are run by the evm_contracts analyzer and won't block the main runtime analyzer, and 2) most of this backfilling will be handled by the migration; the upserts should be relatively rare and inexpensive if we catch the contracts early. (assuming the postgres query optimizer is smart enough to skip the `with` clause)

Reply:
Thank you for the measurements :ok_hand: . I also think a few seconds is fine; there are only so many very popular contracts, and we'll run the expensive query only once for each of them, i.e. when we first download their bytecode.

We do have to run its equivalent for _all_ contracts when we'll first deploy this PR and the db migration will be applied ... and that will probably take minutes or even tens of minutes, during which time the analyzer will be frozen. But I'd say that's fine also, given the low traffic we're getting for now.